src/id128.rs

```diff
@@ -105,6 +105,21 @@ pub fn get_machine_app_specific(app_id: &Id128) -> Result<Id128, SdError> {
     machine_id.app_specific(app_id)
 }
 
+/// Return the unique ID of this boot.
+pub fn get_boot() -> Result<Id128, SdError> {
+    let mut buf = String::new();
+    let mut fd = fs::File::open("/proc/sys/kernel/random/boot_id")
```

Comment:
The reason why I did not add this feature initially is that this requires `procfs`, which can be mounted at a different mountpath other than `/proc` (but it's normally there).
So ideally I wanted this method to take an existing file descriptor to the procfs mounpoint, and I've been waiting for https://doc.rust-lang.org/stable/std/os/unix/io/struct.BorrowedFd.html to stabilize before doing that.

If you need it now, I think we can land this with the caveat that I'd possibly change/break its signature in the not-very-distant future.

Reply:
I think systemd requires procfs mounted at /proc, doesn't it?

Reply:
Yes, systemd itself does require/setup that on the host.
The concern here was more about softwares that do chroot() or create mount namespaces (e.g. daemon and container runtimes). Those may (temporarily) run without procfs or with an uncommon mountpoint.

But I agree with the underlying reasoning, let's add support for this now so that the most common cases are covered.